clients/src/main/java/org/apache/kafka/common/utils/ByteBufferOutputStream.java

```diff
@@ -105,7 +105,7 @@ public int initialCapacity() {
         return initialCapacity;
     }
 
-    private void maybeExpandBuffer(int remainingRequired) {
+    public void maybeExpandBuffer(int remainingRequired) {
```

Comment:
The issue is that we expose access to the buffer directly, but not the ability to expand it. That makes it difficult to use in scenarios where we need to write directly to the underlying buffer. Do you think this is an invalid use case?